perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_attribute_def_def_o365EmailAddresses_o365mu.java

```diff
@@ -0,0 +1,142 @@
+		AttributesManagerBl am = sess.getPerunBl().getAttributesManagerBl();
+		try {
+			if (emails.isEmpty()) {
+				Attribute adName = am.getAttribute(sess, group, ADNAME_ATTRIBUTE);
+				if (adName.getValue() != null) {
+					throw new WrongReferenceAttributeValueException(attribute, adName, group, null, group, null, "at least one email must be defined");
+				}
+
+				return;
+			}
+		} catch (AttributeNotExistsException e) {
+			throw new InternalErrorException(e.getMessage(), e);
+		}
+
+		//check for duplicities among all members and groups
+		AttributesManagerBl attributesManagerBl = sess.getPerunBl().getAttributesManagerBl();
+		Set<Integer> setOfGroupsDefinedByAttribute = attributesManagerBl.getPerunBeanIdsForUniqueAttributeValue(sess, attribute).stream().map(Pair::getLeft).collect(Collectors.toSet());
+		setOfGroupsDefinedByAttribute.remove(group.getId());
+
+		if (!setOfGroupsDefinedByAttribute.isEmpty()) {
+			throw new WrongReferenceAttributeValueException(attribute, attribute, group, null, "some of the email addresses are already assigned to the following group pairs: " + setOfGroupsDefinedByAttribute);
+		}
+		Attribute memberO365EmailAddresses = new Attribute(new urn_perun_member_attribute_def_def_o365EmailAddresses_mu().getAttributeDefinition());
+		memberO365EmailAddresses.setValue(emails);
+		Set<Pair<Integer, Integer>> memberPairs = attributesManagerBl.getPerunBeanIdsForUniqueAttributeValue(sess, memberO365EmailAddresses);
+		if (!memberPairs.isEmpty()) {
+			throw new WrongReferenceAttributeValueException(attribute, memberO365EmailAddresses, group, null, "member " + BeansUtils.getSingleId(memberPairs) + " ");
+		}
+	}
+	/**
+	 * Prefills value created by joining value of urn:perun:groupe:attribute-def:def:adName:o365mu with "@group.muni.cz"
+	 */
+	@Override
+	public Attribute fillAttribute(PerunSessionImpl sess, Group group, AttributeDefinition attrDef) throws WrongAttributeAssignmentException {
+		if (!AttributesManager.NS_GROUP_ATTR_DEF.equals(attrDef.getNamespace())) throw new WrongAttributeAssignmentException(attrDef);
+		try {
+			Attribute result = new Attribute(attrDef);
+			String adName = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, group, ADNAME_ATTRIBUTE).valueAsString();
+			if (adName != null) {
+				result.setValue(Lists.newArrayList(adName + "@group.muni.cz"));
+			}
+			return result;
+		} catch (AttributeNotExistsException e) {
+			throw new InternalErrorException(e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public List<String> getDependencies() {
+		return Collections.singletonList(ADNAME_ATTRIBUTE);
+	}
+
+	@Override
+	public AttributeDefinition getAttributeDefinition() {
+		AttributeDefinition attr = new AttributeDefinition();
+		attr.setNamespace(AttributesManager.NS_GROUP_ATTR_DEF);
+		attr.setFriendlyName("o365EmailAddresses:o365mu");
+		attr.setDisplayName("MU O365 email addresses in o365mu");
+		attr.setType(ArrayList.class.getName());
+		attr.setUnique(true);
```

Comment:
Judging by the commit message this attribute is not mentioned as ``unique`` by definition, you already check duplicates in ``checkAttributeSyntax()``. Or you can have it unique and then remove that check, but it will prevent two different groups to have same email (not just unique within one attribute value), which I'm not sure is wanted or not.

Also do we care about case-insensitive or non-accented uniqueness too (remembering the problems with tcsMails:mu attribute)?

Reply:
It is my mistake. The attribute is already defined as unique and it will stay in this format even for the group.
Because we just need to copy the existing version of module to the new one, there is no need to fix all situations like "case sensitivity" etc.